### ISSUE TITLE
How to use $refs proxy?

### ISSUE THREAD

**Issue by nugetchar** [alpinejs/alpine#46]
Let's say I have the following HTML:

```html
<body x-data="{task: {id: '', label: ''}}">
    <section id="add-task-container">
        <form x-on:submit.prevent>
            <input type="text" x-model="task.label">
            <button type="submit" x-on:click="$refs.listTasks.appendChild(taskToDOM(task, $refs.taskTemplate))">Add task</button>
        </form>
    </section>
    <section id="list-tasks-container" x-ref="listTasks">

    </section> 

    <template id="task-template" x-ref="taskTemplate">
        <section class="task">
            <input type="checkbox" class="task-check">
            <p></p>
        </section>        
    </template>
</body>
```

When trying, in my `taskToDOM` function, to do something like this:
```javascript
function taskToDOM(task, template) {
    const div = template.content.querySelector('.task');
    const node = document.importNode(div, true)
    .....
}  
```
I have got the following error:
`Uncaught TypeError: Illegal invocation`

I am not very used to JavaScript Proxies, so maybe I am doing something wrong?

**Comment by calebporzio** [comment]
No, you are NOT doing something wrong.

I tried fixing this problem the other day. It's a really deep, hacky issue with JavaScript not having the proper scoping inside a `with()` statement (which is what Alpine uses to evaluate the JS expressions in directives)

I'd love help digging in on it.

**Comment by calebporzio** [comment]
Ok, I dug in and found it was an issue with $refs being accessed as a recursive Proxy - ugh.

All fixed now in 1.3.1
